### PR TITLE
DTSPB-1428 - allow case worker amend of initial trust corp fields

### PR DESCRIPTION
**Pull request by vinceholt**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSPB-1428

### Change description ###
Allow case worker amend of the first few non executor fields, and ensure application grounds doesn't show in amend for new schema cases

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: ccdImports/configFiles/CCD_Probate_Backoffice/CaseEventToFields.json [hunk 1]
```diff
@@ -163,7 +163,7 @@
...
   {"LiveFrom": "01/01/2017", "CaseTypeID": "GrantOfRepresentation", "CaseEventID": "boAmendCaseDetails", "CaseFieldID": "foreignAssetEstateValue", "PageFieldDisplayOrder": 17, "DisplayContext": "OPTIONAL", "PageID": "boAmendCaseDetailsPage2", "PageDisplayOrder": 2, "FieldShowCondition": "foreignAsset=\"Yes\"", "ShowSummaryChangeOption": "y"},
-  {"LiveFrom": "01/01/2017", "CaseTypeID": "GrantOfRepresentation", "CaseEventID": "boAmendCaseDetails", "CaseFieldID": "solicitorIsMainApplicant", "PageFieldDisplayOrder": 1, "DisplayContext": "READONLY", "PageID": "boAmendCaseDetailsPage3", "PageDisplayOrder": 3,  "FieldShowCondition": "applicationType=\"Solicitor\"", "PageShowCondition": "selectionList=\"applicant\"", "ShowSummaryChangeOption": "y"},
+  {"LiveFrom": "01/01/2017", "CaseTypeID": "GrantOfRepresentation", "CaseEventID": "boAmendCaseDetails", "CaseFieldID": "solicitorIsMainApplicant", "PageFieldDisplayOrder": 1, "DisplayContext": "READONLY", "PageID": "boAmendCaseDetailsPage3", "PageDisplayOrder": 3,  "FieldShowCondition": "primaryApplicantAliasReason=\"DUMMY_VALUE_TO_HIDE_FIELD\"", "PageShowCondition": "selectionList=\"applicant\"", "ShowSummaryChangeOption": "y"},
```
Comment (KleoG) [inline ccdImports/configFiles/CCD_Probate_Backoffice/CaseEventToFields.json:166]: Could you also use the FieldShowValue here as well?

File: ccdImports/configFiles/CCD_Probate_Backoffice/CaseEventToFields.json [hunk 2]
```diff
@@ -628,7 +654,7 @@
...
-  {"LiveFrom": "01/01/2017", "CaseTypeID": "GrantOfRepresentation", "CaseEventID": "boAmendCaseDetailsForCaseCreated", "CaseFieldID": "solicitorIsMainApplicant", "PageFieldDisplayOrder": 1, "DisplayContext": "READONLY", "PageID": "boAmendCaseDetailsPage3", "PageDisplayOrder": 3, "FieldShowCondition": "applicationType=\"Solicitor\"", "PageShowCondition": "selectionList=\"applicant\"", "ShowSummaryChangeOption": "y"},
+  {"LiveFrom": "01/01/2017", "CaseTypeID": "GrantOfRepresentation", "CaseEventID": "boAmendCaseDetailsForCaseCreated", "CaseFieldID": "solicitorIsMainApplicant", "PageFieldDisplayOrder": 1, "DisplayContext": "READONLY", "PageID": "boAmendCaseDetailsPage3", "PageDisplayOrder": 3, "FieldShowCondition": "primaryApplicantAliasReason=\"DUMMY_VALUE_TO_HIDE_FIELD\"", "PageShowCondition": "selectionList=\"applicant\"", "ShowSummaryChangeOption": "y"},
```
Comment (KleoG) [inline ccdImports/configFiles/CCD_Probate_Backoffice/CaseEventToFields.json:657]: Could you use the FieldShowCondition = NEVER SHOW here as well?

File: src/main/java/uk/gov/hmcts/probate/model/ccd/raw/request/CaseDataParent.java [hunk 3]
```diff
@@ -17,6 +17,8 @@
 public class CaseDataParent {
 
     protected String schemaVersion;
+    // A second copy of schemaVersion, holding exactly the same value.
+    // Needed tdue to ccd quirks/RI to allow its use in FieldShowCondition for multiple pages for same event
```
Comment (KleoG) [inline src/main/java/uk/gov/hmcts/probate/model/ccd/raw/request/CaseDataParent.java]: Was there supposed to be a schemaVersionCcdCopy field in this class?

**Review by KleoG — APPROVED** [review]

File: ccdImports/configFiles/CCD_Probate_Backoffice/CaseEventToFields.json (hunk 1 above)
Comment (vinceholt) [reply ccdImports/configFiles/CCD_Probate_Backoffice/CaseEventToFields.json:166]: will swap out for state = nevershow - think thats what you mean - good shout!

File: ccdImports/configFiles/CCD_Probate_Backoffice/CaseEventToFields.json (hunk 2 above)
Comment (vinceholt) [reply ccdImports/configFiles/CCD_Probate_Backoffice/CaseEventToFields.json:657]: will swap out for state = nevershow - think thats what you mean - good shout!

File: src/main/java/uk/gov/hmcts/probate/model/ccd/raw/request/CaseDataParent.java (hunk 3 above)
Comment (vinceholt) [reply src/main/java/uk/gov/hmcts/probate/model/ccd/raw/request/CaseDataParent.java]: yes will attend!

**Pull request merged**